### PR TITLE
Implement Athena get_database and list_databases APIs, closing #9899

### LINKED ISSUE #9899: Missing implementation for Athena: get_database
I have a setup with terraform using a local moto_server to test deploy the infrastructure (and then hopefully afterwards directly test functionality against that infrastructure). However, when deploying an `aws_athena_database` resource, it failed, because the get_database call apparently does not exist:

```
File "/home/dev01/Documents/my/project/.venv/lib/python3.11/site-packages/moto/core/responses.py", line 623, in call_action
    raise NotImplementedError(f"The {action} action has not been implemented")
NotImplementedError: The get_database action has not been implemented
```

**python and moto version**
```
$ python --version
Python 3.11.2

$ pip list | grep moto
moto                      5.1.22
```

I quickly looked at the athena code and saw it does not exist there. The API call itself does exist in the API description: https://docs.aws.amazon.com/athena/latest/APIReference/API_Operations.html

Am I missing something or is it indeed missing?
If the implementation for this is relatively simple, I would like to take a crack at it. I've got moto setup locally.

### PR DESCRIPTION
**Pull request by rajendersaini**
Closes #9899
 
 ## Summary
 Implements `get_database` and `list_databases` for the Athena service, matching real AWS behavior.
 
 ## Changes
 - **`moto/athena/exceptions.py`** — Added `MetadataException` (HTTP 400, raised when database not found)
 - **`moto/athena/models.py`** — Added `Database` model, DDL parsing in `start_query_execution` for `CREATE/DROP DATABASE|SCHEMA`, 
`get_database()` and `list_databases()` with pagination
 - **`moto/athena/responses.py`** — Added `get_database` and `list_databases` response handlers
 - **`tests/test_athena/test_athena_database.py`** — 14 tests covering all operations and edge cases
 - **`other_langs/terraform/athena/`** — Terraform example for `aws_athena_database`
 - **`IMPLEMENTATION_COVERAGE.md`** — Marked `get_database` and `list_databases` as implemented (35% → 38%)
 
 ## Behavior
 - Databases created via `CREATE DATABASE` / `CREATE SCHEMA` DDL in `start_query_execution`
 - Database names lowercased (matches AWS/Glue behavior)
 - `default` database pre-populated under `AwsDataCatalog`
 - `IF NOT EXISTS` / `IF EXISTS` supported
 - Catalog context from `QueryExecutionContext.Catalog` respected
 - `MetadataException` raised for nonexistent databases (matches AWS error format)
 
 ## Checklist
 - [x] Feature added
 - [x] Linter passes (`ruff check`, `ruff format`, `mypy`)
 - [x] Tests added (14 new tests)
 - [x] All existing tests pass (59 passed, 1 skipped)

### CODE REVIEW

File: moto/athena/models.py [hunk 1]
```diff
@@ -291,10 +318,99 @@ def start_query_execution(
...
+                f"Error Code: EntityNotFoundException)"
+            )
+        db = self.databases[key]
+        return {
+            "Name": db.name,
+            "Description": db.description,
+            "Parameters": db.parameters,
+        }
+
+    def list_databases(
```
Comment (bblommers) [inline moto/athena/models.py]: Can we use the `@paginator` utility here? That would avoid writing our own custom logic to make this work.

There are quite a few examples in the codebase already if you want some inspiration, and the user docs are here: https://docs.getmoto.org/en/latest/docs/contributing/development_tips/utilities.html#paginator

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @rajendersaini, thanks for the PR! Looks good overall - nice to see the Terraform test built-in as well.

Just have one request to simplify the pagination feature, if we can change that, I'd be happy to merge it.

**Comment by rajbayer** [comment]
@bblommers  - can you take a another look ?

**Review by bblommers — APPROVED** [review]
Great, thank you @rajendersaini!
